### PR TITLE
Added Warning header and unit tests.

### LINKED ISSUE #355: Add support for Warning headers?
It would be useful to support:

https://tools.ietf.org/html/rfc7234#section-5.5

### PR DESCRIPTION
**Pull request by andrey-qlogic**
Fixes #355

- [x] Tests pass
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: google-http-client/src/main/java/com/google/api/client/http/HttpHeaders.java [hunk 1]
```diff
@@ -811,6 +815,37 @@ public HttpHeaders setAuthenticate(String authenticate) {
     return this;
   }
 
+  /**
+   * Sets the {@code "Warning"} header or {@code null} for none.
+   *
+   * <p>Overriding is only supported for the purpose of calling the super implementation and
```
Comment (elharo) [inline google-http-client/src/main/java/com/google/api/client/http/HttpHeaders.java]: Is this enforced in code? Could it be, perhaps with a helper method that's protected and making this method final? But that wouldn't allow you to change the return type I suppose.

File: google-http-client/src/main/java/com/google/api/client/http/HttpHeaders.java [hunk 2]
```diff
@@ -811,6 +815,37 @@ public HttpHeaders setAuthenticate(String authenticate) {
...
+  public final String getWarning() {
+    return getFirstHeaderValue(warning);
+  }
+
+  /**
+   * Returns all {@code "Warning"} headers or {@code null} for none.
+   *
+   * @since 1.28
+   */
+  public final List<String> getWarningAsList() {
```
Comment (elharo) [inline google-http-client/src/main/java/com/google/api/client/http/HttpHeaders.java]: Does this expose potentially mutable internal state?

File: google-http-client/src/main/java/com/google/api/client/http/HttpHeaders.java [hunk 3]
```diff
@@ -811,6 +815,37 @@ public HttpHeaders setAuthenticate(String authenticate) {
...
+  /**
+   * Sets the {@code "Warning"} header or {@code null} for none.
+   *
+   * <p>Overriding is only supported for the purpose of calling the super implementation and
+   * changing the return type, but nothing else.
+   *
+   * @since 1.28
+   */
+  public HttpHeaders setWarning(String warning) {
+    this.warning = getAsList(warning);
```
Comment (elharo) [inline google-http-client/src/main/java/com/google/api/client/http/HttpHeaders.java]: Can there be more than one of these? The code seems undecided.

File: google-http-client/src/main/java/com/google/api/client/http/HttpHeaders.java (hunk 1 above)
Comment (andrey-qlogic) [reply google-http-client/src/main/java/com/google/api/client/http/HttpHeaders.java]: Done.

File: google-http-client/src/main/java/com/google/api/client/http/HttpHeaders.java [hunk 4]
```diff
@@ -811,6 +815,35 @@ public HttpHeaders setAuthenticate(String authenticate) {
...
 
+  /**
+   * Sets the {@code "Warning"} header or {@code null} for none.
+   *
+   * <p>Overriding is only supported for the purpose of calling the super implementation and
+   * changing the return type, but nothing else.
+   *
+   * @since 1.28
+   */
+  public HttpHeaders setWarning(String warning) {
```
Comment (elharo) [inline google-http-client/src/main/java/com/google/api/client/http/HttpHeaders.java]: Looks like `addWarning` would be more accurate than `setWarning`.

Comment (andrey-qlogic) [reply google-http-client/src/main/java/com/google/api/client/http/HttpHeaders.java]: @elharo , thanks for the comment, renamed method.

**Comment by andrey-qlogic** [comment]
@sduskis , PTAL

File: google-http-client/src/main/java/com/google/api/client/http/HttpHeaders.java [hunk 5]
```diff
@@ -158,6 +158,10 @@ public HttpHeaders() {
   @Key("User-Agent")
   private List<String> userAgent;
 
+  /** {@code "Warning"} header. */
+  @Key("Warning")
+  private List<String> warning;
```
Comment (sduskis) [inline google-http-client/src/main/java/com/google/api/client/http/HttpHeaders.java]: please rename this variable to `warnings`.

Comment (andrey-qlogic) [reply google-http-client/src/main/java/com/google/api/client/http/HttpHeaders.java]: Renamed

**Pull request merged**